File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/Substance.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Substance : DomainResource, IResource
{
    public required Base.CodeableConcept Code { get; set; }
    public string? Status { get; set; }
    public Base.CodeableConcept[]? Category { get; set; }
    public Substance_Instance[]? Instance { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Substance_Ingredient[]? Ingredient { get; set; }
    public string? Description { get; set; }

    public class Substance_Instance : BackboneElement
    {
        public string? Expiry { get; set; }
        public Base.Quantity? Quantity { get; set; }
        public Base.Identifier? Identifier { get; set; }
    }

    public class Substance_Ingredient : BackboneElement
    {
        public Base.Ratio? Quantity { get; set; }
        public object? Substance    
        {
            get
            {
                if (SubstanceReference is not null)
                {
                    return SubstanceReference;
                }
        
                if (SubstanceCodeableConcept is not null)
                {
                    return SubstanceCodeableConcept;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    SubstanceReference = (Base.ResourceReference)value;return;
                }
        
                if (value?.GetType() == typeof(Base.CodeableConcept))
                {
                    SubstanceCodeableConcept = (Base.CodeableConcept)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.ResourceReference? SubstanceReference { get; set; }
        public Base.CodeableConcept? SubstanceCodeableConcept { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/Substance.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import Quantity
from base import DomainResource
from base import Ratio
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class SubstanceInstance(BackboneElement):
    expiry: Optional[str] = None
    quantity: Optional[Quantity] = None
    identifier: Optional[Identifier] = None

@dataclass(kw_only=True)
class SubstanceIngredient(BackboneElement):
    quantity: Optional[Ratio] = None
    substance_reference: Optional[Reference] = None
    substance_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class Substance(DomainResource):
    code: CodeableConcept
    status: Optional[str] = None
    category: Optional[List[CodeableConcept]] = field(default_factory=list)
    instance: Optional[List[SubstanceInstance]] = field(default_factory=list)
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    ingredient: Optional[List[SubstanceIngredient]] = field(default_factory=list)
    description: Optional[str] = None